diagnostics-extension/src/controllers/events.ts

```diff
@@ -6,14 +6,88 @@
  * @fileoverview Handle event requests
  */
 
-import { Request, Response } from '@common/message';
-import { generateErrorResponse } from '../utils';
+import { EventCategory } from '@common/dpsl';
+import {
+  EventsAction,
+  Request,
+  Response,
+  ResponseErrorInfoMessage,
+} from '@common/message';
+import { EventsServiceProvider } from '../services/events';
+import {
+  generateErrorResponse,
+  generateEventsSuccessResponse,
+} from '../utils';
 
 type EventsController = (
   category: Request,
   res: (data: Response) => void
 ) => void;
 
-export const handleEvents: EventsController = (req, res) => {
-  res(generateErrorResponse('Events not implemented.'));
+const eventsService = EventsServiceProvider.getEventsService();
+
+const isEventSupported = async (
+  eventType: EventCategory,
+  res: (data: Response) => void,
+) => {
+  try {
+    const paylaod = await eventsService.isEventSupported(eventType);
+    return res(generateEventsSuccessResponse(paylaod));
+  } catch (err) {
+    res(generateErrorResponse(String(err)));
+  }
+}
+
+const startCapturingEvents = async (
+  eventType: EventCategory,
+  res: (data: Response) => void,
+) => {
+  try {
+    await eventsService.startCapturingEvents(eventType);
+    const payload = {};
+    return res(generateEventsSuccessResponse(payload));
+  } catch (err) {
+    res(generateErrorResponse(String(err)));
+  }
+}
+
+const stopCapturingEvents = async (
+  eventType: EventCategory,
+  res: (data: Response) => void,
+) => {
+  try {
+    await eventsService.stopCapturingEvents(eventType);
+    return res(generateEventsSuccessResponse({}));
+  } catch (err) {
+    res(generateErrorResponse(String(err)));
+  }
+}
+
+export const registerEventHandlers = async () => {
```

Comment:
Per comment in `sw.ts`

Let's not export these two functions. 

Instead export just one function called `onEventPortConnect`, where we will do all the registration within events.ts

Reply:
Sure. Done.